### PR TITLE
Fixes #33842 - Make ordering by id possible for more models

### PR DESCRIPTION
**Pull request by ofedoren**
Not sure if I added for all models, but mainly for those which have scoped search by name. I'm adding this due to an old BZ [1]. The BZ is more about wrong description of --order option in hammer, but since the description comes from the API docs I've decided to look whether we even have possibility to order users by id, which we don't.

[1] - https://bugzilla.redhat.com/show_bug.cgi?id=1881668

### CODE REVIEW

**Review by tbrisker — COMMENTED** [review]
Should we move this definition to `ApplicationRecord`?

**Comment by ofedoren** [comment]
@tbrisker, well, that would make more sense to have this in one place instead of repeating stuff. I just wanted to be more "safe", since I'm not sure we have `id` column for every `ApplicationRecord`. I'm aware that Rails generate those by default, but you can always drop that column if you don't need it and you prefer other uniq key.

**Comment by tbrisker** [comment]
That is a fair concern, I was wondering how many of these we have so I ran the following on my dev box which has a bunch of plugins installed:
```
ActiveRecord::Base.connection.tables.reject{|table| ActiveRecord::Base.connection.columns(table).any?{|col| col.name == 'id'}} 
=> ["ar_internal_metadata",
 "dynflow_execution_plans",
 "dynflow_delayed_plans",
 "dynflow_schema_info",
 "foreman_openscap_asset_policies",
 "locations_organizations",
 "media_operatingsystems",
 "operatingsystems_provisioning_templates",
 "operatingsystems_ptables",
 "operatingsystems_puppetclasses",
 "schema_migrations",
 "architectures_operatingsystems"]
```
So it would seem the only such tables we have (we can ignore the dynflow and rails internal ones) are simple join tables, which I think don't even have models defined and thus should be hopefully safe. We might want to run the same on a box with katello to make sure they don't have such models though.

File: app/models/auth_sources/auth_source_ldap.rb [hunk 1]
```diff
@@ -41,8 +41,6 @@ class AuthSourceLdap < AuthSource
   before_validation :sanitize_use_netgroups
   after_initialize :set_defaults
 
-  scoped_search :on => :name, :complete_value => :true
```
Comment (ezr-ondrej) [inline app/models/auth_sources/auth_source_ldap.rb]: I don't like this in the same commit

**Comment by ezr-ondrej** [comment]
> since I'm not sure we have `id` column for every `ApplicationRecord`

I believe we can just have guard on `id` column exists? we would need to be careful about the DB being ready, but it should not be much of an issue.

**Comment by ofedoren** [comment]
@tbrisker, I've checked with Katello installed and few other plugins and I've got the exact same list you had.

@ezr-ondrej, I also changed the PR in a way that we now ensure each model will have scope search by id if it was inherited from `ApplicationRecord`, which is mostly models defined by us.

Should I also remove older definitions from the models? It shouldn't affect anything, but the number of lines of code.

**Comment by ezr-ondrej** [comment]
> Should I also remove older definitions from the models? It shouldn't affect anything, but the number of lines of code.

Yep please, otherwise we will not clean it up later on :)

**Comment by ofedoren** [comment]
@ezr-ondrej, updated. It looks like there was nothing much to remove :D I didn't remove this line https://github.com/theforeman/foreman/blob/eb5d51a9325de745236a5f929f6b5ca51b56d532/app/models/concerns/audit_search.rb#L18 since this module is being included in `Audit`, which is being created explicitly here: https://github.com/theforeman/foreman/blob/eb5d51a9325de745236a5f929f6b5ca51b56d532/config/initializers/audit.rb#L9

**Comment by ezr-ondrej** [comment]
[test katello]
Is probably not related

**Review by ezr-ondrej — DISMISSED** [review]
Thanks @ofedoren ! LGTM :+1:

**Comment by ezr-ondrej** [comment]
Oh the Katello failures are actually related... Those resources allow string ids...

**Comment by ofedoren** [comment]
@ezr-ondrej, well, after some digging, I've noticed that this "general" way won't work (or at least without any complications): if we set scoped search on `ApplicationRecord`, all inheriting models will have it, which is quite the point, but in some cases it breaks functionality e.g. for https://github.com/Katello/katello/blob/30e6fe7d91d6296bb02830e4afaa3f66bee38795/app/models/katello/package_group.rb#L9, since our scoped search definition on `id` column will take precedence over that scoped search with the same name (`rename: :id`) probably because of https://github.com/wvanbergen/scoped_search/blob/10d118690b6ba5a4941ac17e3d9157ddc817592f/lib/scoped_search/definition.rb#L90 + https://github.com/wvanbergen/scoped_search/blob/10d118690b6ba5a4941ac17e3d9157ddc817592f/lib/scoped_search/definition.rb#L217.

Due to that, the only way I see is to return that buch of manually defined searches for each model at least for Foreman :/

**Review by ezr-ondrej — APPROVED** [review]
Thanks @ofedoren, I do not agree with what katello does, I believe we should avoid that, but unfortunately it is what we are working with. Sorry to push you the wrong path before.
LGTM now :+1:

**Pull request merged**